crates/dies-simulator/src/lib.rs

```diff
@@ -339,7 +443,291 @@ impl Simulation {
         self.feedback_queue.pop_front()
     }
 
+    pub fn update_game_state(&mut self) {
+        // Only print when the state changes
+        static mut LAST_STATE: Option<SimulationGameState> = None;
+        let should_print = unsafe {
+            match LAST_STATE {
+                //  Some is part of the Option enum, which is used to represent a value 
+                // that can be either present (Some(value)) or absent (None).
+                Some(last) if self.game_state.equal(&last) => false,
+                _ => {
+                    LAST_STATE = Some(self.game_state);
+                    true
+                }
+            }
+        };
+        if should_print {
+            self.game_state.println(&self.current_time);
+        }
+
```

Comment:
we should never use unsafe in the codebase as that breaks all the nice guarantees rust gives. lets discuss 1:1 how to do this properly